homework6.js

```diff
@@ -0,0 +1,212 @@
+// ---------
+// 1a)
+// ---------
+
+var Ctor = function() {
+  this.a = 0;
+  this.b = 1;
+}
+
+var obj1 = new Ctor();
+var obj2 = new Ctor();
+
+var obj3 = {};
+
+Ctor.call(obj3);
+
+obj1.c = 2;
+
+obj1.constructor.prototype.d = 3;
+
+// obj1 and obj2 will both have a property d equal to 3 because we set that
+// property on the prototype.
+// obj3 however, will not have a d property because we did not use the
+// new keyward when creating object 3, and so it does not share that prototype
+
+// ---------
+// 1b)
+// ---------
+
+// the difference is that for objA we overrode the prototype and set it equal
+// to a new object with properties of num and str. With objB we simply added
+// two properties to it's original prototype. So objA has a prototype of Object
+// and objB has a prototype of B (with an extra property constructor
+// referenceing the function that created it)
+
```

Comment:
Yep! See Dan's answer below for a more in depth explanation:

The prototype which is created automatically when its constructor is defined
is given a property 'constructor' which is then inherited by the constructor's instances.

So objB.constructor is B.

But replacing the default prototype with a new object, as with

A.prototype = ...

doesn't reset the constructor property on the new object.   Therefore objA inherits 
the new prototype's original constructor, which is Object.

So objA.constructor is Object, not A.